jax/_src/callback.py

```diff
@@ -129,25 +135,50 @@ def callback_batching_rule(
     args,
     dims,
     *,
-    vectorized: bool,
+    vectorized: bool | None | DeprecatedArg,
+    vmap_method: str | None,
     result_avals: Sequence[core.ShapedArray],
     **kwargs: Any,
 ):
-  axis_size = next(a.shape[d] for a, d in zip(args, dims)
-                   if d is not batching.not_mapped)
+  if isinstance(vectorized, DeprecatedArg) and vmap_method is None:
+    warnings.warn(
```

Comment:
Should we add a TODO to drop this in 6 months?

Also, it would be nice to use `deprecations.warn` here so that we could accelerate the deprecation for Google-internal code.

Reply:
Done and done. Thanks for the suggestions!